packages/react-slider/Spec.md

```diff
@@ -1,63 +1,232 @@
+| buttonProps       | Additional props for the actual `role="slider"` (slider box) element. (Note that this element is not actually a button in the current implementation.) | Personally not sure if it is essential but up for discussion                                 |
+| valueFormat       | Custom formatter for the slider value. value                                                                                                           | Label will handle custom formats                                                             |
+| originFromZero    | Whether to attach the origin of slider to zero. Helpful when the range include negatives. @defaultvalue false                                          | Replaced with origin to allow for more control over the component.                           |
+| ranged            | Whether to render a **Ranged Slider**. Ranged Sliders display two `thumbs` that allow for lower and upper bounds to be easily selected.                | Ranged is enabled when a value or defaultValue pass an array with a left and right position. |
 
 ## Structure
 
 - _**Public**_
+  ```jsx
+  <Slider label={label} marks={marks} />
+  ```
 - _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
 
-## Migration
+  ```jsx
+  <slots.root {...slotProps.root}>
+    {state.label && <slots.label {...slotProps.label} />}
+    <slots.sliderContainer {...slotProps.sliderContainer}>
+      <slots.rail {...slotProps.rail} />
+        {state.marks && (
+            <slots.markContainer {...slotProps.markContainer}>
+              <slots.mark {...slotProps.mark} />
+              {state.markLabel && <slots.markLabel {...slotProps.markLabel} />}
+           </slots.markContainer>
+      <slots.track {...slotProps.track} />
+      <slots.thumb {...slotProps.thumb} />
+    </slots.sliderContainer>
+    {state.valueLabel && <slots.valueLabel {...slotProps.valueLabel} />}
+  </slots.root>
+  ```
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+- _**DOM** - how the component will be rendered as HTML elements_
 
-- _Migration from v8_
-- _Migration from v0_
+```jsx
+    <div className="ms-Slider-root">
+        // Label is up for discussion
+        <div className="ms-Slider-label">
+            {...label children}
+        </div>
+        <div className="ms-Slider-container">
+            <div className="ms-Slider-rail" />
+            <div className="ms-Slider-markContainer">
+                <div className="ms-Slider-mark" />
+                <div className="ms-Slider-markLabel" />
+            </div>
+            <div className="ms-Slider-track" />
+            <div className="ms-Slider-thumb" />
+        </div>
+        <div className="ms-Slider-valueLabel" />
+    </div>
+```
```

Comment:
I would specially like to see what @smhigley thinks of the structure proposed here since I know that we might not want everything to be a div due to accessibility purposes (maybe we need to render a non-visible input).